torch_xla/csrc/convolution.cpp

```diff
@@ -10,6 +10,89 @@
 
 namespace torch_xla {
 namespace {
+// clang-format off
+/* Lower PyTorch Conv & its backward to XLA
+ * This file covers lowerings of both forward and backward of conv op.
+ *   - BuildConvolutionOverrideable
+ *     - BuildTransposedConvolution
+ *     - Normal(non-transpose) convolution
+ *   - BuildConvolutionBackwardOverrideable
+ *     - BuildConvBackwardInput
+ *     - BuildConvBackwardWeight
+ *     - BuildGradBias
+ *
+ * Here're detailed steps from a 4D PyTorch inputs to inputs calling into
+ * ConvGeneralDilated (the most general conv op in XLA).
+ * PyTorch input format & shape:
+ *   - input: [N, Cin, Hin, Win]
+ *   - weight: [Cout, Cin / groups, KH, KW]
+ *   - output: [N, Cout, Hout, Wout]
+ *
+ * Output, grad_input and grad_weight can all be calculated as
+ * ConvGeneralDilated with proper transpose/padding etc.
+ *   - output: conv(input, weight)
+ *   - grad_input: conv(grad_output, weight^T) (with padding etc)
+ *   - grad_weight: conv(input^T, grad_output)
+ *
+ * XLA provides the following wrappers instead of calling into raw
+ * ConvGeneralDilated.
+ * tensorflow/tensorflow/compiler/tf2xla/kernels/conv_op_helpers.cc
+ *   - MakeXlaForwardConvOp (not used in our lowering, see below)
+ *   - MakeXlaBackpropInputConvOp
+ *   - MakeXlaBackpropFilterConvOp
+ *
+ * Shapes below use format in XLA implementation.
+ * For group(non-depthwise) convolutions(G > 1, Cout = M * G):
```

Comment:
What is `M`?

Reply:
`M` is `multiplier`, same as `channel_multiplier` here. https://www.tensorflow.org/api_docs/python/tf/nn/depthwise_conv2d 
Pytorch doesn't have this term since it's not exposed as expected input size. In XLA lowering it's sometimes used.

Reply:
I'd mention that in your comment.